### ISSUE TITLE
Terraform Initial Review 

### ISSUE THREAD

**Issue by cgriggs01** [akamai/terraform-provider-akamai#16]
Hello Akamai Team,

My name is Chris Griggs, I am a Partner Solutions Engineer here a HashiCorp and will be your main point of contact as we review and release your provider. 

To Start, it appears that your provider does not have any website documentation. You will need to write documentation for your provider and each of its resources.

Additionally, HashiCorp runs nightly acceptance tests on each official provider. You will need to provide access credentials and/or end-point for which we can test against. The credentials and be securely submitted via [web-form](https://docs.google.com/forms/d/e/1FAIpQLSe5TjsThekgY0nUZan8M8JFVFk2qFC0shR6Fwpggn8XhhoHdg/viewform?usp=sf_link).

If you have any questions please feel free to reach out.

Best,
Chris.

**Comment by dshafik** [comment]
@cgriggs01 we previously submitted the credentials to Clint a while back (almost a year to the day), and he likely would be the only person with easy access to the account at this point.

I'll get started on documentation this week.

**Comment by cgriggs01** [comment]
Thanks @dshafik, got the creds!

**Comment by sherodtaylor** [comment]
Is there an ETA on the official release of this? We are thinking about using this but not sure whether to wait :)

**Comment by cgriggs01** [comment]
@dshafik How is the documentation going., do you have an ETA on the completion of the docs?

@sherodtaylor This provider should be released in the next few months, once the documentation is complete HashiCorp will start reviewing the provider.

**Comment by ninnemannk** [comment]
Documentation would be fantastic.

**Comment by dshafik** [comment]
@cgriggs01 why was this closed? What is the status of reviewing the provider?

**Comment by cgriggs01** [comment]
Hi @dshafik,

Sorry, I sent you an email last week, but should have reach out though this. 

Whats the reasoning for implementing the `property` resource in eight different files for each method, was this for organizational purposes or did you have another reason? 

I will reopen this issue now. 

Thanks
Chris

**Comment by cgriggs01** [comment]
Hi @dshafik 

Hope things are going well, I'd like to move this provider to the next phase of the program so we can officially release. I'm getting customer feedback that this provider would be useful. 

The way that the `property` resource is implemented does not follow the terraform provider file format requirements. Can you move all your resource methods in one resource file?(minus the tests methods) If you have any reason for obligating to this, please let me know an we can talk though it. 

Best,
Chris

**Comment by dshafik** [comment]
@cgriggs01 this was an attempt to organize such a large of code (comparatively). It's quite hairy in there. So, in-as-much as it's just a stylistic choice, I'm fine to merge it if that's the recommendation.

**Comment by dshafik** [comment]
@cgriggs01 the merge has been completed now

**Comment by cgriggs01** [comment]
@dshafik It definitely helps to make the property resource much more organized, but it is a requirement so that all official Terraform providers follow the structure for continuity across providers when reviewing and attempting to debug a provider. 

Thank you for making the changes.

**Comment by mb290** [comment]
@dshafik and @cgriggs01 is this still actively being reviewed and being made "officially supported"?

**Comment by dshafik** [comment]
@mb290 absolutely

**Comment by tfhartmann** [comment]
We are also interested in seeing this made an officially supported provider.  Thank you everyone for all the work you're doing on this provider.

**Comment by derekargueta** [comment]
ping

**Comment by nitrocode** [comment]
Yes, thank you for supporting terraform and akamai integration. I'm eagerly waiting official support and documentation on how to use this.

Any update on documentation?
